packages/astro/src/core/render/dev/index.ts

```diff
@@ -150,11 +150,19 @@ export async function render(
 
 	let styles = new Set<SSRElement>();
 	[...stylesMap].forEach(([url, content]) => {
-		// The URL is only used by HMR for Svelte components
-		// See src/runtime/client/hmr.ts for more details
+		// Vite handles HMR for styles injected as scripts
+		scripts.add({
+			props: {
+				type: 'module',
+				src: url,
+				'data-astro-injected': true,
```

Comment:
I think this isn't needed any more.

Reply:
These `data-astro-injected` props are still referenced in `astro/client/hmr.js`